infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -323,122 +356,111 @@ private KubernetesNamespaceMeta asNamespaceMeta(Namespace namespace) {
     return new KubernetesNamespaceMetaImpl(namespace.getMetadata().getName(), attributes);
   }
 
-  /**
-   * Creates a Kubernetes namespace for the specified workspace.
-   *
-   * <p>The namespace name will be chosen according to a configuration, and it will be prepared
-   * (created if necessary).
-   *
-   * @param workspaceId identifier of the workspace
-   * @return created namespace
-   * @throws InfrastructureException if any exception occurs during namespace preparing
-   */
-  public KubernetesNamespace create(String workspaceId) throws InfrastructureException {
-    final String namespaceName;
-    try {
-      namespaceName = evalNamespaceName(workspaceId, EnvironmentContext.getCurrent().getSubject());
-    } catch (NotFoundException | ServerException | ConflictException | ValidationException e) {
-      throw new InfrastructureException(
-          format(
-              "Failed to determine the namespace to put the workspace %s to."
-                  + " The error message was: %s",
-              workspaceId, e.getMessage()),
-          e);
-    }
+  public KubernetesNamespace getOrCreate(RuntimeTarget target) throws InfrastructureException {
```

Comment:
the name of this method is confusing. I would expect it to get the namespace if it exists or create a new one and return if it does not. But it looks like it's doing something a bit different. Could you please at least document the method and maybe think of better name?

Reply:
The thing is that this will return a namespace and it will also create it if configured to do so. So it will get it or create and get it.